### ISSUE TITLE
Help with triggering MQTT payload on switch toggle to on

### ISSUE THREAD

**Issue by kiwijunglist** [arendst/Sonoff-Tasmota#4561]
```
23:17:34 MQT: stat/shelly2.frontdoor/STATUS = {"Status":{"Module":47,"FriendlyName":["Sonoff","Sonoff2"],"Topic":"shelly2.frontdoor","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS1 = {"StatusPRM":{"Baudrate":4800,"GroupTopic":"sonoffs","OtaUrl":"http://sonoff.maddox.co.uk/tasmota/sonoff.bin","RestartReason":"Power on","Uptime":"19T16:49:45","StartupUTC":"2018-11-18T05:27:49","Sleep":0,"BootCount":58,"SaveCount":343,"SaveAddress":"F5000"}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS2 = {"StatusFWR":{"Version":"6.2.1.8","BuildDateTime":"2018-09-26T23:00:44","Boot":31,"Core":"2_3_0","SDK":"1.5.3(aec24ac9)"}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["manitoba",""],"TelePeriod":300,"SetOption":["00008009","55818000","00000000"]}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS4 = {"StatusMEM":{"ProgramSize":477,"Free":524,"Heap":15,"ProgramFlashSize":1024,"FlashSize":2048,"FlashMode":3,"Features":["00000809","0FDAE794","000003A0","23B617CE","000003C0"]}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS5 = {"StatusNET":{"Hostname":"shelly2-frontdoor","IPAddress":"192.168.1.75","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer":"192.168.1.50","Mac":"84:F3:EB:9F:61:16","Webserver":2,"WifiConfig":5}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.50","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_9F6116","MqttUser":"mike","MqttType":1,"MAX_PACKET_SIZE":1000,"KEEPALIVE":15}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS7 = {"StatusTIM":{"UTC":"Fri Dec 07 22:17:34 2018","Local":"Fri Dec 07 23:17:34 2018","StartDST":"Sun Mar 25 02:00:00 2018","EndDST":"Sun Oct 28 03:00:00 2018","Timezone":1,"Sunrise":"08:29","Sunset":"16:53"}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS9 = {"StatusPTH":{"PowerDelta":80,"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS10 = {"StatusSNS":{"Time":"2018-12-07T23:17:34","Switch1":"ON","Switch2":"OFF","ENERGY":{"Total":4.453,"Yesterday":0.503,"Today":0.402,"Power":0,"Factor":0.00,"Voltage":0,"Current":0.000,"Frequency":0}}}
23:17:34 MQT: stat/shelly2.frontdoor/STATUS11 = {"StatusSTS":{"Time":"2018-12-07T23:17:34","Uptime":"19T16:49:45","Vcc":3.145,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"manitoba","BSSId":"78:8A:20:49:4C:8B","Channel":11,"RSSI":76}}}
```

Hi

I have a shelly 2 running tasmota wired to 2x lights.  I use external toggle switches and I use home assistant for my hub.  I want another light elsewhere in the house to turn on when the light wired to the shelly2 is turned on using the toggle switch.  I don't want the light to turn on when the shelly2 light is triggered via MQTT.

So basically I want an MQTT message to be sent from tasmota when the light is turned on via the switch (but not when turned on via MQTT).  If this is not possible then I want an MQTT message whenever the switch is toggled, and I can use an additional automation in home assistant to check whether the shelly2 light is on or off.

I tried using rule on Switch1#state=ON/toggle/1 do publish cmnd/sonoff.rubbish_light/POWER ON endon , but it didn't seem to do anything. example with toggle below.

```
23:32:22 CMD: rule1 on switch1#state=toggle do publish cmnd/sonoff.rubbish_light/POWER ON endon
23:32:22 SRC: WebConsole from 192.168.1.52
23:32:22 RSL: Group 0, Index 1, Command RULE, Data on switch1#state=toggle do publish cmnd/sonoff.rubbish_light/POWER ON endon
23:32:22 MQT: stat/shelly2.frontdoor/RESULT = {"Rule1":"ON","Once":"OFF","StopOnError":"OFF","Free":436,"Rules":"on switch1#state=toggle do publish cmnd/sonoff.rubbish_light/POWER ON endon"}
23:32:24 CMD: rule1 on
23:32:24 SRC: WebConsole from 192.168.1.52
23:32:24 RSL: Group 0, Index 1, Command RULE, Data on
23:32:24 MQT: stat/shelly2.frontdoor/RESULT = {"Rule1":"ON","Once":"OFF","StopOnError":"OFF","Free":436,"Rules":"on switch1#state=toggle do publish cmnd/sonoff.rubbish_light/POWER ON endon"}
23:32:32 SRC: Switch
23:32:32 MQT: stat/shelly2.frontdoor/RESULT = {"POWER1":"ON"}
23:32:32 MQT: stat/shelly2.frontdoor/POWER1 = ON
23:32:33 SRC: Switch
23:32:33 MQT: stat/shelly2.frontdoor/RESULT = {"POWER1":"OFF"}
23:32:33 MQT: stat/shelly2.frontdoor/POWER1 = OFF
23:32:34 SRC: Switch
23:32:34 MQT: stat/shelly2.frontdoor/RESULT = {"POWER2":"ON"}
23:32:34 MQT: stat/shelly2.frontdoor/POWER2 = ON
23:32:35 SRC: Switch
23:32:35 MQT: stat/shelly2.frontdoor/RESULT = {"POWER2":"OFF"}
23:32:35 MQT: stat/shelly2.frontdoor/POWER2 = OFF
```

This is probably quite an easy problem, so sorry I am unable to do it myself.

Thanks for any help you can give me!

Cheers
Mike

**Comment by ascillato** [comment]
Hi, your rule approach is the right path if you  want your switch not to toggle also your local output. You need just to correct a little the syntax.

rule on Switch1#state do publish cmnd/sonoff.rubbish_light/POWER ON endon

If you want to toggle also the local output, you can use instead:

rule on Switch1#state do backlog power1 2; publish cmnd/sonoff.rubbish_light/POWER ON endon

Hope this helps. 

For support it is faster the Tasmota Support Chat. Github issues are meant for software bug reports

**Comment by ascillato2** [comment]
Closing this issue as it has been answered.

### Support Information
See [Wiki](https://github.com/arendst/Sonoff-Tasmota/wiki) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.

**Comment by kiwijunglist** [comment]
```
00:36:59 CMD: Switch1#state do backlog power1 2; publish cmnd/sonoff.rubbish_light/POWER ON endon
00:36:59 SRC: WebConsole from 192.168.1.52
00:36:59 RSL: Group 0, Index 1, Command SWITCH1#STATE, Data do backlog power1 2; publish cmnd/sonoff.rubbish_light/POWER ON endon
00:36:59 MQT: stat/shelly2.frontdoor/RESULT = {"Command":"Unknown"}
```

unknown command error

**Comment by ascillato** [comment]
Hehehe. To store a rule you need to put the complete syntax

Go to the console and type

Rule1 1

Rule1 on Switch1#state do backlog power1 2; publish cmnd/sonoff.rubbish_light/POWER ON endon

**Comment by kiwijunglist** [comment]
oh stupid me

**Comment by kiwijunglist** [comment]
let me try again :D

**Comment by kiwijunglist** [comment]
ok it works for toggle the switch by sending the MQTT message whenever the light is turned on (or off via the toggle switch).

Is there a way that i only send the MQTT message if the toggle switch is turning on the light, and don't send MQTT if the light is switched off via the toggle switch?

**Comment by ascillato** [comment]
Yes,

Switchmode1 1 
Rule1 1
Rule1 on Switch1#state=1 do backlog power1 2; publish cmnd/sonoff.rubbish_light/POWER 2 endon on Switch1#state=0 do delay endon

**Comment by kiwijunglist** [comment]
that kinda works but it turns my toggle switch into a momentary push button switch for controlling the light, so you can't operate the light ordinary, eg i have to flip the switch twice to toggle the light.

**Comment by ascillato** [comment]
Change switchmode to 7

**Comment by ascillato** [comment]
Another even better option is:

Switchmode1 1
Rule1 1
Rule1 on Switch1#state=1 do backlog power1 1; publish cmnd/sonoff.rubbish_light/POWER 1 endon on Switch1#state=0 do backlog power1 0; publish cmnd/sonoff.rubbish_light/POWER 0 endon

**Comment by kiwijunglist** [comment]
Solution:

Tasmota Console:
```
switchmode2 0
rule1 1
rule1 on switch2#state do backlog power2 2; publish stat/shelly2.frontdoor/TOGGLE2 ON endon
```

Automation.yaml:
```
- alias: "turn on the rubbish light if front door light is turned on via the switch"
  trigger:
    platform: mqtt
    topic: stat/shelly2.frontdoor/TOGGLE2
    payload: 'ON'
  condition:
    - condition: state
      entity_id: light.front_door
      state: 'on'    
  action:
    - service: light.turn_on
      entity_id: light.rubbish_light
```

**Comment by kiwijunglist** [comment]
Better solution from ascillato:

```
Switchmode2 0
Rule1 1
Rule1
   on power2#boot do var2 %value% endon
   on power2#state do var2 %value% endon
   on Switch2#state do backlog power2 2; event sendcommand=%var2% endon
   on event#sendcommand=0 do publish cmnd/sonoff.rubbish_light/POWER 1 endon
```
